app/classes/query/base_new.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+# base class for Query searches
+class Query::BaseNew
+  include ActiveModel::API
+  include ActiveModel::Attributes
+  include ActiveModel::Validations::Callbacks
+
+  include Query::Modules::ClassMethods
+  include Query::ScopeModules::ResultAccessors
+  include Query::ScopeModules::Initialization
+  include Query::ScopeModules::SequenceOperators
+  include Query::Modules::Validation
+
+  validates_with Query::Modules::Validator
+
+  # "clean up" and reassign attributes before validation
+  before_validation :clean_and_validate_params
+
+  attr_writer :record
+
+  def self.parameter_declarations
+    { order_by: :string }
+  end
+
+  delegate :parameter_declarations, to: :class
+
+  # can use has_attribute? here.
+  # only called in ApplicationController::Queries#apply_one_content_filter
+  def self.takes_parameter?(key)
+    parameter_declarations.key?(key)
+  end
+
+  delegate :takes_parameter?, to: :class
+
+  def self.scope_parameters
+    excepts = [:id_in_set, :preference_filter]
+    @scope_parameters = parameter_declarations.except(*excepts).keys +
+                        [:id_in_set]
```

Comment:
Reads like you are first removing `:id_in_set` as a valid param, then adding it back.

Reply:
That's right. 

Oh, wait... I think this is an artifact of when i was trying to `compact` the params _other than_  `id_in_set`: that is the only param that should preserve an empty array as its value, because it forces empty results. 

I'll review and check if i can remove this, thanks.

Reply:
Oh i know. It's to force order. `:id_in_set` must come last because it reorders results, but to ensure that it unsets any previous order it must be last in the chain.

Reply:
Added comment explaining